Cabal/src/Distribution/Simple/Build.hs

```diff
@@ -928,6 +932,61 @@ replFLib flags pkg_descr lbi exe clbi =
         GHC -> GHC.replFLib flags NoFlag pkg_descr lbi exe clbi
         _ -> dieWithException verbosity REPLNotSupported
 
+-- | Runs 'componentInitialBuildSteps' on every configured component.
+--
+-- Legacy function: does not run pre-build hooks or pre-processors. This function
+-- is insufficient on its own to prepare the build for a package.
+--
+-- Consumers wanting to prepare the sources of a package, e.g. in order to
+-- launch a REPL session, are advised to run @Setup repl --repl-multi-file=<fn>@
+-- instead.
```

Comment:
My only thought - and it is not a 'blocker' - is that `Setup repl --repl-multi-file=<fn>` could have greater consistency with the Cabal User Guide, which is written in terms of:

    runhaskell Setup.hs repl --repl-multi-file=<fn>

If that is applicable, it is also applicable below.

Reply:
Let me try to understand: `Setup` assumes the setup script is compiled (with something like `ghc --make Setup.hs`, I guess?) and the binary is invoked, while `runhaskell Setup.hs` interprets it and thus runs is? Is that it? Do both of these forms work so that it's just a user choice? If so, I'd rather avoid creating a new PR on master to change that (the original PR is already merged), unless @mpilgrem feels strongly one of the forms is misleading or bad style. But if one of these is actually wrong, please let me know and I will create the PR myself (and let's also fix here in that case).

Reply:
My review comment was a subtle one, and I do not feel strongly about it. I was exploring the following:

What does it mean to advise: `run @Setup repl --repl-multi-file=<fn>@`?

It means (as [documented in the Cabal User Guide](https://cabal.readthedocs.io/en/stable/setup-commands.html) for other uses of the Cabal Haskell 'script' file named (by convention) `Setup.hs`) to command `runhaskell Setup.hs --repl-multi-file=<fn>` (or, equivalently, `runghc Setup.hs --repl-multi-file=<fn>`). My suggestion was to state things more plainly in the Haddock documentation and in a way consistent with the User Guide.